sample-xml/app/src/main/java/co/nimblehq/sample/xml/di/modules/PreferencesModule.kt

```diff
@@ -0,0 +1,32 @@
+package co.nimblehq.sample.xml.di.modules
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.preferencesDataStore
+import co.nimblehq.sample.xml.data.repository.AppPreferencesRepositoryImpl
+import co.nimblehq.sample.xml.domain.repository.AppPreferencesRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+private val Context.appPreferencesDataStore by preferencesDataStore("app_preferences")
+
+@Module
+@InstallIn(SingletonComponent::class)
+class PreferencesModule {
+
+    @Provides
+    @Singleton
```

Comment:
@lydiasama I don't think we need to set `DataStore<Preferences>` as Singleton. It is enough to put it inside a `SingletonComponent` module to keep its lifespan. Please check this for more detail: https://medium.com/digigeek/hilt-components-and-scopes-in-android-b96546cb07df

> Scoping is expensive. Use it if you really need it !!

So, please consider this issue for XML & Compose samples or templates too 🤗

Reply:
@luongvo Please correct me if I'm wrong but from your reference, it seems we still need to add it.

> Component determines the scope of the instance. But when we ask Hilt to give us an OkHttpClient in two different places, will it be singleton or not ? By default, it is not singleton. If you want an application wide singleton OkHttpClient instance, you should annotate your OkHttpClient provider method with @Singleton.

Reply:
@lydiasama's idea is kind of like the implementation from [this thread](https://stackoverflow.com/a/66101769). There is an updated version from 2022 and from Google developer's [article](https://medium.com/androiddevelopers/datastore-and-dependency-injection-ea32b95704e3). Since we are using hilt and providing the dependency, I think we should go with the builder pattern instead of delegate pattern here. What do you guys think? cc @luongvo

Reply:
Yes, I agree with @kaungkhantsoe. That seems to be a better + newer approach to apply. cc @lydiasama. 

Regarding my concern about using `@Singleton`. You're right, it seems we still need to keep only 1 instance of DataStore for safety. 🙏